web/src/main/java/se/inera/intyg/rehabstod/service/certificate/CertificateServiceImpl.java

```diff
@@ -156,6 +175,88 @@ public GetLUCertificatesForCareUnitResponse getLUCertificatesForCareUnit(GetLUCe
         return new GetLUCertificatesForCareUnitResponse(luCertificateList, qaInfoError);
     }
 
+    /**
+     * This method will replace getLUCertificatesForCareUnit(request). For backward compatibility both methods will
+     * coexist until the new MetaData table in Intygstjansten is in full use. This method can be renamed and the old method
+     * removed from 2021-2 and after.
+     */
+    private GetLUCertificatesForCareUnitResponse getNewLUCertificatesForCareUnit(GetLUCertificatesForCareUnitRequest request) {
+        final var user = userService.getUser();
+        final var unitIds = user.getValdVardenhet().getHsaIds();
+
+        final var certTypesToQuery = getCertificateTypesToQuery(request.getCertTypes());
+        final var doctorIds = getDoctorIds(user, request.getDoctors());
+
+        var diagnosedCertificateList = restIntegrationService
+            .getDiagnosedCertificatesForCareUnit(unitIds, certTypesToQuery, request.getFromDate(), request.getToDate(), doctorIds);
+
+        diagnosedCertificateList = filterOnDiagnoses(diagnosedCertificateList, request.getDiagnoses());
+        diagnosedCertificateList = filterOnAge(diagnosedCertificateList, request.getFromAge(), request.getToAge());
+
+        puService.enrichDiagnosedCertificateWithPatientNamesAndFilterSekretess(diagnosedCertificateList);
+
+        var luCertificateList = newTransformDiagnosedCertificatesToLUCertificates(diagnosedCertificateList);
+
+        final var qaInfoError = populateLUCertificatesWithNotificationData(luCertificateList);
+
+        luCertificateList = filterOnQuestionAndAnswers(request, luCertificateList);
+
+        luCertificateList = filterOnText(luCertificateList, request.getSearchText());
+
+        pdlLogLUCertificatesForCareUnit(luCertificateList);
+
+        LOGGER.debug("Returning LU Certificates for Care Unit");
+        return new GetLUCertificatesForCareUnitResponse(luCertificateList, qaInfoError);
+    }
+
+    private List<DiagnosedCertificate> filterOnDiagnoses(List<DiagnosedCertificate> diagnosedCertificates, List<String> diagnoses) {
+        if (diagnoses == null || diagnoses.isEmpty()) {
+            return diagnosedCertificates;
+        }
+
+        return diagnosedCertificates.stream()
+            .filter(c -> filterOnDiagnoses(c, diagnoses))
+            .collect(Collectors.toList());
+    }
+
+    private List<DiagnosedCertificate> filterOnAge(List<DiagnosedCertificate> diagnosedCertificates, int fromAge, int toAge) {
+        if (fromAge > 0 || (toAge > 0 && toAge <= 100)) {
```

Comment:
Går det att skapa en konstant som förklarar gränsen på 100?

Reply:
Har lagt in konstanter för lower och upper age limits.